x-pack/plugins/maps/public/classes/sources/vector_source/vector_source.tsx

```diff
@@ -0,0 +1,215 @@
+}
+
+export interface BoundsFilters {
+  applyGlobalQuery: boolean;
+  filters: Filter[];
+  query?: MapQuery;
+  sourceQuery?: MapQuery;
+  timeFilters: TimeRange;
+}
+
+export interface IVectorSource extends ISource {
+  getTooltipProperties(properties: GeoJsonProperties): Promise<ITooltipProperty[]>;
+  getBoundsForFilters(
+    boundsFilters: BoundsFilters,
+    registerCancelCallback: (callback: () => void) => void
+  ): Promise<MapExtent | null>;
+  getGeoJsonWithMeta(
+    layerName: string,
+    searchFilters: VectorSourceRequestMeta,
+    registerCancelCallback: (callback: () => void) => void,
+    isRequestStillActive: () => boolean
+  ): Promise<GeoJsonWithMeta>;
+
+  getFields(): Promise<IField[]>;
+  getFieldByName(fieldName: string): IField | null;
+  getLeftJoinFields(): Promise<IField[]>;
+  getSyncMeta(): VectorSourceSyncMeta | null;
+  getFieldNames(): string[];
+  getApplyGlobalQuery(): boolean;
+  createField({ fieldName }: { fieldName: string }): IField;
+  canFormatFeatureProperties(): boolean;
+  getSupportedShapeTypes(): Promise<VECTOR_SHAPE_TYPE[]>;
+  isBoundsAware(): boolean;
+  getSourceTooltipContent(sourceDataRequest?: DataRequest): SourceTooltipConfig;
+}
+
+export interface ITiledSingleLayerVectorSource extends IVectorSource {
+  getUrlTemplateWithMeta(
+    searchFilters: VectorSourceRequestMeta
+  ): Promise<{
+    layerName: string;
+    urlTemplate: string;
+    minSourceZoom: number;
+    maxSourceZoom: number;
+  }>;
+  getMinZoom(): number;
+  getMaxZoom(): number;
+  getLayerName(): string;
+}
+
+export class AbstractVectorSource extends AbstractSource implements IVectorSource {
+  static async getGeoJson({
+    format,
+    featureCollectionPath,
+    fetchUrl,
+  }: {
+    format: FORMAT_TYPE;
+    featureCollectionPath: string;
+    fetchUrl: string;
+  }) {
+    let fetchedJson;
+    try {
+      const response = await fetch(fetchUrl);
+      if (!response.ok) {
+        throw new Error('Request failed');
+      }
+      fetchedJson = await response.json();
+    } catch (e) {
+      throw new Error(
+        i18n.translate('xpack.maps.source.vetorSource.requestFailedErrorMessage', {
+          defaultMessage: `Unable to fetch vector shapes from url: {fetchUrl}`,
+          values: { fetchUrl },
+        })
+      );
+    }
+
+    if (format === FORMAT_TYPE.GEOJSON) {
+      return fetchedJson;
+    }
+
+    if (format === FORMAT_TYPE.TOPOJSON) {
+      const features = _.get(fetchedJson, `objects.${featureCollectionPath}`);
+      return topojson.feature(fetchedJson, features);
+    }
+
+    throw new Error(
+      i18n.translate('xpack.maps.source.vetorSource.formatErrorMessage', {
+        defaultMessage: `Unable to fetch vector shapes from url: {format}`,
+        values: { format },
+      })
+    );
+  }
+
+  getFieldNames(): string[] {
+    return [];
+  }
+
+  createField({ fieldName }: { fieldName: string }): IField {
+    throw new Error('Not implemented');
+  }
+
+  /**
```

Comment:
Can you remove this comment. With TS these comments are duplicated and in this case, actually wrong since there is no label parameter and return is IField | null